app/controllers/OrganisationDetailsController.scala

```diff
@@ -46,20 +47,32 @@ trait OrganisationDetailsController extends LisaBaseController {
           Future.successful(BadRequest(views.html.registration.organisation_details(formWithErrors)))
         },
         data => {
-          cache.cache[OrganisationDetails](cacheId, OrganisationDetails.cacheKey, data)
+          cache.fetchAndGetEntry[BusinessStructure](cacheId, BusinessStructure.cacheKey).flatMap { bStructure =>
+            Logger.debug("BusinessStructure retrieved")
+            rosmService.rosmRegister(bStructure.get.businessStructure, data).flatMap(res =>
+              res.isRight match {
```

Comment:
Rather than matching on res.isRight, can it instead be a pattern match on Right(safeId) and Left(error), will remove the need for the res.right.get and res.left.get too.